### PR TITLE
language: analyzeEntitySentiment support

### PR DESCRIPTION
**Pull request by callmehiphop**
This adds support for the `analyzeEntitySentiment` rpc.

### CODE REVIEW

**Comment by stephenplusplus** [comment]
Is this going to replace `detectSentiment`?

File: packages/language/src/document.js [hunk 1]
```diff
@@ -531,6 +529,86 @@ Document.prototype.detectEntities = function(options, callback) {
 };
 
 /**
+ * Detect the entities from a block of text, similar to
+ * {module:language/document#detectEntities}, and analyzes sentiment associated
```
Comment (callmehiphop) [inline packages/language/src/document.js]: Done

File: packages/language/src/index.js [hunk 2]
```diff
@@ -234,6 +235,92 @@ Language.prototype.detectEntities = function(content, options, callback) {
 };
 
 /**
+ * Detect the entities from a block of text, similar to
+ * {module:language#detectEntities}, and analyzes sentiment associated with
```
Comment (callmehiphop) [inline packages/language/src/index.js]: Done

**Comment by stephenplusplus** [comment]
- detectEntitySentiment = detect entities; include their sentiment scores
- detectEntities = detect entities; don't include their sentiment scores

Would things be simpler if we could just use an option to distinguish between intent, as opposed to multiple methods?

```js
language.detectEntities({ sentiment: false }) // default
language.detectEntities({ sentiment: true }) // calls detectEntitySentiment
```

Better yet, how about we just *always* include sentiment? Is there something expensive about this method that it needs to be separated from the one that doesn't include sentiment? The API response is exactly the same, with the exception of the `sentiment` object being present or not.

**Comment by callmehiphop** [comment]
> Better yet, how about we just always include sentiment? Is there something expensive about this method that it needs to be separated from the one that doesn't include sentiment? The API response is exactly the same, with the exception of the sentiment object being present or not.

@stephenplusplus I think it would be better to get some feedback from googlers on this.

/cc @monattar @landrito @lukesneeringer

**Comment by monattar** [comment]
Wrt question about why not always including sentiment: This feature is more expensive than all other features. We shouldn't include the sentiment by default, as it will incur high costs that the user might not intend.

We thought quite a bit about where to expose this API (within analyzeEntity, as a separate method, etc), and separate method has been the best. Please do not change the surface.

**Comment by stephenplusplus** [comment]
Thanks for the information.

> and separate method has been the best.

Why?

**Comment by lukesneeringer** [comment]
We do not want to merge this for a couple of reasons.

  * We do not want to opt people in to the 1.1 beta API. Users should get v1 unless they explicitly request otherwise.
  * We have decided to only expose the new method on the GAPIC for now (and DPEs are handling documenting this).

Sorry, @callmehiphop -- I did not realize you were working on this last week, or I would have updated you sooner.

**Pull request closed without merging**